### PR TITLE
do not save edgecore token on disk

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
do not save edgecore token on disk
```

### CODE REVIEW

**Comment by wackxu** [comment]
I think we should not store the token in any file, this token is only used when the edge node first join, for example

```
keadm join --token=xxxx ...
```

so the token could only happen in the command, and edgecore will use the certificate to access the cloudcore

so we can just remove the token filed from the edgecore config

**Comment by gy95** [comment]
just for some backward compatibility, e.g.  use v1.13 keadm  to install v1.12 edgecore, we should keep `token` for some releases and then delete it.

File: edge/cmd/edgecore/app/server.go [hunk 1]
```diff
@@ -62,6 +65,31 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
...
+
+					// cleanup token
+					if err := cleanupToken(*config, opts.ConfigFile); err != nil {
+						klog.Exit(err)
+					}
+				}()
+			}
+
+			// Get token from bootstrap file if it exists
+			bootstrapFile := "/etc/kubeedge/bootstrap-edgecore.conf"
```
Comment (wackxu) [inline edge/cmd/edgecore/app/server.go:84]: set this variable as a constant

File: edge/cmd/edgecore/app/server.go [hunk 2]
```diff
@@ -62,6 +65,31 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
...
+			if utilvalidation.FileIsExist(bootstrapFile) {
```
Comment (wackxu) [inline edge/cmd/edgecore/app/server.go:85]: if the file does not exist, we should throw an err and exitt

Comment (gy95) [reply edge/cmd/edgecore/app/server.go:85]: if edgecore restart, bootstrap file also not exist, we should not throw an error

Comment (gy95) [reply edge/cmd/edgecore/app/server.go:92]: add a pre check

File: edge/cmd/edgecore/app/server.go (hunk 1 above)
Comment (gy95) [reply edge/cmd/edgecore/app/server.go]: fixed

**Comment by kevin-wangzefeng** [comment]
security improvement
/cc @vincentgoat

**Review by kevin-wangzefeng — COMMENTED** [review]
There's a failure of CI fuzzing check, please take a look, thanks

File: keadm/cmd/keadm/app/cmd/edge/join.go [hunk 3]
```diff
@@ -232,6 +270,9 @@ func createEdgeConfigFiles(opt *common.JoinOptions) error {
 	}
 
 	edgeCoreConfig.Modules.EdgeHub.WebSocket.Server = opt.CloudCoreIPPort
+	// TODO: remove this after release 1.14
+	// this is for keeping backward compatibility
+	// don't save token in configuration edgecore.yaml
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/edge/join.go:275]: Let's also add a `depracated` comment in token field, then rermove it in future release.

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/edge/join.go:275]: if we run edgecore diretly, i think we can allow users to fill in token in edgecore.yaml, or users should write this token to /etc/kubeedge/bootstrap-edgecore.conf, this is not easy use

Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/edge/join.go:275]: If user run edgecore diretly, they should prepare the certs by themselfs. And token field in config file is not safe, we must remove it in future. cc @vincentgoat How do you think?

Comment (vincentgoat) [reply keadm/cmd/keadm/app/cmd/edge/join.go:277]: Agree. The token should be constrained into the file /etc/kubeedge/bootstrap-edgecore.conf for temp use.

File: keadm/cmd/keadm/app/cmd/edge/join.go [hunk 4]
```diff
@@ -403,3 +432,15 @@ func runEdgeCore() error {
...
+func createBootstrapFile(opt *common.JoinOptions) error {
+	bootstrapFile := constants.BootstrapFile
+	_, err := os.Create(bootstrapFile)
+	if err != nil {
+		return err
+	}
+
+	// write token to bootstrap-edgecore.conf file
+	token := []byte(opt.Token)
+	return os.WriteFile(bootstrapFile, token, 0666)
```
Comment (vincentgoat) [inline keadm/cmd/keadm/app/cmd/edge/join.go:445]: `0640` is the proper permission for the file.

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/edge/join.go]: fixed

File: pkg/apis/componentconfig/edgecore/v1alpha2/types.go [hunk 5]
```diff
@@ -297,6 +297,7 @@ type EdgeHub struct {
 	// Optional if quic is configured
 	WebSocket *EdgeHubWebSocket `json:"websocket,omitempty"`
 	// Token indicates the priority of joining the cluster for the edge
+	// Deprecated: will be removed after release 1.14, will not be saved in configuration file
```
Comment (fisherxu) [inline pkg/apis/componentconfig/edgecore/v1alpha2/types.go:300]:
```suggestion
	// Deprecated: will be removed in future release, will not be saved in configuration file
```
Need more discussion for removing in which one release :)

Comment (gy95) [reply pkg/apis/componentconfig/edgecore/v1alpha2/types.go]: yes, fixed

**Review by fisherxu — COMMENTED** [review]
/appove
/lgtm

**Pull request merged**